icefoganalytics/travel-authorization
# Standardize `expenses` table in Database

References
- https://github.com/icefoganalytics/travel-authorization/issues/16
- https://github.com/icefoganalytics/travel-authorization/issues/30
- [Modeling, My Travel Requests, 2023-10-13](https://docs.google.com/document/d/1jpW3VokZa3v5Mv-Gf4Kfjve2WwTpK7X4cpep5-EROCE/edit?pli=1#heading=h.epyy9mt07rdp)

# Context

1. Snake case/underscore all `expenses` columns
2. Fix all foreign key references in the table so that they refer to the table name they are referencing (i.e. `taid`).
3. Add `created_at` and `update_at` standard timestamp columns.
4. Migrate Expense#type column so that it references Model names. e.g. Expense or Estimate and update all references to the column so that the references uses these constants.

# Implementation

Update `taid` fake foreign key, and make it a real foreign key to the `forms` table as `form_id`
Start exposing enums through the Model instead of independent export, because I was worried about naming conflicts, and namespacing to the model avoids that problem.
Migrate all code querying the `expenses` table using legacy Knex queries, to using new Sequelize model queries.

# Testing Instructions

1. Boot the db and api services via `dev up`.
2. Boot the web service by `dev up_web`.
3. Check that the app boots correctly.
5. Go to the My Travel Request page.
6. Create a new travel request.
7. Create some to/from travel information and save the draft.
8. Go to the "Estimate" tab and generate an estimate.
9. Check that you can create, edit, and delete estimates.
